main.py

```diff
@@ -9,9 +9,207 @@
 * We suggest using a convolutional neural network.
 * TensorFlow Keras has the CIFAR-10 dataset as a module, so you don't need to manually download and unpack it.
 """
-
 # Import whatever libraries/modules you need
 
+import torch
 import numpy as np
+import matplotlib.pyplot as plt
+import torch.optim as optim
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision
+import torchvision.transforms as transforms
+from torchvision import datasets, transforms
 
 # Your working code here
+"""
+CIFAR-10 has 10 classes of images, and each image is 32 x 32 pixels and has 3 color values. 
+Training data set size is 50,000 images, and test dataset size is 10,000 images.
+"""
+# Transforming the images
+"""
+- transforms.Compose() takes in a sequence of transformations that are to be applied to the input data.
+- transforms.ToTensor() converts the input images to tensors with the shape of channels, height, and width. 
+- transforms.Normalize() takes in two arguments, mean and std. Since CIFAR10 is a 3 channel dataset, (0.5, 0.5, 0.5)
+represents the mean values and standard deviation for the RGB channels. 
+"""
+
+transform_data = transforms.Compose(
+    [transforms.ToTensor(), transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
+
+# Loading the testing and training data sets
+# Batch size is the number of samples processed in one iteration, ie, forward and backward pass.
+batch_size = 16
+
+# Loading the dataset
+train_set = torchvision.datasets.CIFAR10(root='/data',
+                                         train=True,
+                                         download=True,
+                                         transform=transform_data)
+
+train_loader = torch.utils.data.DataLoader(train_set,
+                                           batch_size=batch_size,
+                                           shuffle=True)
+
+test_set = torchvision.datasets.CIFAR10(root='/data',
+                                        train=False,
+                                        download=True,
+                                        transform=transform_data)
+
+test_loader = torch.utils.data.DataLoader(test_set,
+                                          batch_size=batch_size,
+                                          shuffle=True)
+
+
+# Developing a Neural Network Architecture as a class
+# Subclassing nn.Module
+class Network (nn.Module):
+    def __init__(self):
+        # Network inherits its functions and behavior from super class nn.Module
+        super(Network, self).__init__()
+        self.conv1 = nn.Conv2d(
+            in_channels=3, out_channels=32, kernel_size=(3, 3))
+        self.conv2 = nn.Conv2d(
```

Comment:
Nit: I'd suggest have a max pooling layer in the middle of the convultion layers